### PR TITLE
Merge default and downloaded config (EXPOSUREAPP-11750)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
Merges default and downloaded CCL configuration.

No new default config available yet - Check tests for verification.

Details: https://github.com/corona-warn-app/cwa-app-tech-spec/blob/4e297fe221d3524214c94c528def000ff89455f5/docs/spec/dcc-ccl.gen.md#calling-a-ccl-api-with-jsonfunctions

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/CCLConfigurationRepository.kt [hunk 1]
```diff
@@ -43,30 +44,30 @@ class CCLConfigurationRepository @Inject constructor(
...
+                Timber.tag(TAG).d("Nothing to update. Keeping old ccl config list")
+                return@updateBlocking this
+            }
 
-            if (rawData != null && newConfig == null) {
-                // parsing failed
+            val downloadedConfigList = downloadedConfigListRaw.tryParseCCLConfigurations() ?: run {
                 updateResult = UpdateResult.FAIL
+                return@updateBlocking this
             }
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/CCLConfigurationRepository.kt:60]: Not related to the actual merging. Just some refactoring.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/CCLConfigurationRepository.kt [hunk 2]
```diff
@@ -43,30 +44,30 @@ class CCLConfigurationRepository @Inject constructor(
...
-                    newConfig
-                }
-
-                false -> {
-                    Timber.tag(TAG).d("Nothing to update. Keeping old ccl config list")
-                    this
-                }
-            }
+            Timber.tag(TAG).d("Saving new config data")
+            downloadedCclConfigurationStorage.save(rawData = downloadedConfigListRaw)
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/CCLConfigurationRepository.kt:63]: We still only store the downloaded config, but we update the `HotDataFlow` with the merged config (see below)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt [hunk 3]
```diff
@@ -8,7 +8,7 @@ import timber.log.Timber
 import java.io.File
 import javax.inject.Inject
 
-class CCLConfigurationStorage @Inject constructor(
+class DownloadedCCLConfigurationStorage @Inject constructor(
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt:11]: Renamed to `DownloadedCCLConfigurationStorage` to make it clear that only the downloaded config is stored, not the merged one.

**Review by mtwalli — APPROVED** [review]
LGTM

**Review by chiljamgossow — COMMENTED** [review]
Nice test!

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt (hunk 3 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt:11]: You are right, it should actuallly be camelCase. However we are using the non-camle case CLL everywhere. I would suggest that if we want to change it, we should open another refactoring PR. What do you think @mtwalli

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt:11]: Ye, @chiljamgossow  is right, we should follow camelCase

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt:11]: See https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/configuration/storage/DownloadedCCLConfigurationStorage.kt:11]: Ok. But let's change it for all classes in a different PR after this one is merged...

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**